source/extensions/common/crypto/BUILD

```diff
@@ -18,6 +18,7 @@ envoy_cc_extension(
         "crypto_impl.h",
         "utility_impl.h",
     ],
+    category = "SOMECAT",
```

Comment:
i dont see any category with either `common` or `crypto`

Reply:
I'm not even sure if this is technically a pluggable extension and whether it should have `envoy_cc_extension`.

Reply:
should it be converted to an `envoy_cc_library` ?

Reply:
@dmitri-d do you have any opinions on this from a Red Hat perspective?

Reply:
No this should be envoy_cc_extension, it is intended to replace SSL implentation, cc @envoyproxy/openssl-dev

Reply:
@lizan I understand it lives under extensions, but what aspect of `envoy_cc_extension` is relevant? Is there an SSL implementation factory here?

Reply:
I like the idea of an extensible crypto; having said that I'm not sure I understand what the plan here is. How is an ssl implementation meant to be plugged in? Is this extension just for the code in the `extensions/common/crypto` package? Is there a plan to add extension points in, for example, `extensions/transport_sockets/tls` package? But then something like `extensions/transport_sockets/tls/context_impl.cc` should probably be a pluggable implementation (BoringSSL and OpenSSL versions are *very* different)?

Reply:
@lizan @htuch im wondering if we can just add an `UNKNOWN` extension category (at least temporarily) or similar

we can have protodoc etc ignore it i reckon

it would be really helpful to move this PR along

Reply:
I agree, let's land this. I honestly still don't grok whey this needs to be under `envoy_cc_extension` (vs. being an `envoy_cc_library`). Maybe add an extension category `DELIBERATELY_OMITTED`?

Reply:
updated